### PR TITLE
wireless/bcm43xxx: enable tx flow control to improve performance

### PR DESCRIPTION
**Pull request by anchao**
## Summary

1. wireless/bcm43xxx: enable tx flow control to improve performance

RX/TX shared free queue on bcmf implementation, if TX occupies the
free queue completely, RX will trigger read abort because it cannot
successfully alloc buffer for the shared free queue. This commit will
limit the sending entries of tx and prevent rx triggering abort

2. Revert "wireless/bcm43xxx: remove unused tx_queue_count"

## Impact

N/A

## Testing

bcm43013 iperf test

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 1]
```diff
@@ -1065,10 +1065,20 @@ struct bcmf_sdio_frame *bcmf_sdio_allocate_frame(FAR struct bcmf_dev_s *priv,
...
+          sbus->tx_queue_count <
+            CONFIG_IEEE80211_BROADCOM_FRAME_POOL_SIZE / 2)
         {
-          nxsem_post(&sbus->queue_mutex);
-          break;
+          if ((entry = bcmf_dqueue_pop_tail(&sbus->free_queue)) != NULL)
+            {
+              if (tx)
+                {
+                  sbus->tx_queue_count += 1;
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1076]:
```suggestion
                  sbus->tx_queue_count++;
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 2]
```diff
@@ -1108,5 +1118,10 @@ void bcmf_sdio_free_frame(FAR struct bcmf_dev_s *priv,
 
   bcmf_dqueue_push(&sbus->free_queue, &sframe->list_entry);
 
+  if (sframe->tx)
+    {
+      sbus->tx_queue_count -= 1;
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1123]:
```suggestion
      sbus->tx_queue_count--;
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c [hunk 3]
```diff
@@ -1065,10 +1065,20 @@ struct bcmf_sdio_frame *bcmf_sdio_allocate_frame(FAR struct bcmf_dev_s *priv,
           DEBUGPANIC();
         }
 
-      if ((entry = bcmf_dqueue_pop_tail(&sbus->free_queue)) != NULL)
+      if (!tx ||
+          sbus->tx_queue_count <
```
Comment (pkarashchenko) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1069]: do we need set to zero `tx_queue_count` during init, or `struct bcmf_sdio_dev_s` use `zalloc` / static allocation?

**Review by pkarashchenko — APPROVED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c (hunk 3 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1069]: tx_queue_count has been initialized to 0
https://github.com/apache/incubator-nuttx/blob/master/drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c#L816

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c (hunk 2 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1123]: Done

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c (hunk 1 above)
Comment (anchao) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.c:1076]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**